go/mysql/server.go

```diff
@@ -22,16 +22,34 @@ import (
 	"fmt"
 	"io"
 	"net"
+	"time"
 
 	log "github.com/golang/glog"
 	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/stats"
 	"github.com/youtube/vitess/go/tb"
 )
 
 const (
 	// DefaultServerVersion is the default server version we're sending to the client.
 	// Can be changed.
 	DefaultServerVersion = "5.5.10-Vitess"
+
+	// timing metric keys
+	connectTimingKey = "Connect"
+	queryTimingKey   = "Query"
+)
+
+var (
+	// Metrics
+	timings    = stats.NewTimings("MysqlServerTimings")
+	connCount  = stats.NewInt("MysqlServerConnCount")
+	connAccept = stats.NewInt("MysqlServerConnAccepted")
+	connSlow   = stats.NewInt("MysqlServerConnSlow")
+
+	// SlowConnectWarnThreshold if non-nil, specifies an amount of time
+	// beyond which a warning is logged to identify the slow connection
+	SlowConnectWarnThreshold *time.Duration
```

Comment:
How about just using:
SlowConnectWarnThreshold time.Duration
and compare to zero to see if you're using it?

Also, I think this should be stored in the Listener, like AllowClearTextWithoutTLS. Would be simpler.